inference-engine/src/inference_engine/os/win/win_shared_object_loader.cpp

```diff
@@ -113,12 +116,80 @@ class SharedObjectLoader::Impl {
     }
 #endif
 
-public:
+    static const char  kPathSeparator = '\\';
+
+    static const char* FindLastPathSeparator(LPCSTR path) {
+        const char* const last_sep = strchr(path, kPathSeparator);
+        return last_sep;
+    }
+
+    std::basic_string<CHAR> GetDirname(LPCSTR path) {
+        auto pos = FindLastPathSeparator(path);
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<CHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
 #ifdef ENABLE_UNICODE_PATH_SUPPORT
+    static const wchar_t* FindLastPathSeparator(LPCWSTR path) {
+        const wchar_t* const last_sep = wcsrchr(path, kPathSeparator);
+        return last_sep;
+    }
+
+    std::basic_string<WCHAR> GetDirname(LPCWSTR path) {
+        auto pos = FindLastPathSeparator(path);
+        if (pos == nullptr) {
+            return path;
+        }
+        std::basic_string<WCHAR> original(path);
+        original[pos - path] = 0;
+        return original;
+    }
+
+    std::basic_string<WCHAR> IncludePluginDirectory(LPCWSTR path) {
```

Comment:
why not `std::wstring`?

Reply:
it was in initial PR, changed to string/wstring